dynamoprojection/options.go

```diff
@@ -0,0 +1,156 @@
+	}
+}
+
+func (o *options) applyTableOption(decorators *decorators) {
+	if o.applyTableOptionFunc != nil {
+		o.applyTableOptionFunc(decorators)
+	}
+}
+
+// WithDecorateGetItem adds a decorator for GetItem operation. The decorator can
+// modify the passed GetItemInput structure and return a slice of request.Option
+// to alter the request prior to its execution.
+func WithDecorateGetItem(
+	dec func(*dynamodb.GetItemInput) []func(*dynamodb.Options),
+) interface {
+	HandlerOption
+	ResourceRepositoryOption
+} {
+	return &options{
+		applyOptionToAdaptorFunc: func(d *decorators) {},
+		applyResourceRepositoryOptionFunc: func(d *decorators) {
+			d.decorateGetItem = dec
+		},
+	}
+}
+
+// WithDecoratePutItem adds a decorator for PutItem operation. The decorator can
+// modify the passed PutItemInput structure and return a slice of request.Option
+// to alter the request prior to its execution.
+func WithDecoratePutItem(
+	dec func(*dynamodb.PutItemInput) []func(*dynamodb.Options),
+) interface {
+	HandlerOption
+	ResourceRepositoryOption
+} {
+	return &options{
+		applyOptionToAdaptorFunc: func(d *decorators) {},
+		applyResourceRepositoryOptionFunc: func(d *decorators) {
+			d.decoratePutItem = dec
+		},
+	}
+}
+
+// WithDecorateDeleteItem adds a decorator for DeleteItem operation. The
+// decorator can modify the passed DeleteItemInput structure and return a slice
+// of request.Option to alter the request prior to its execution.
+func WithDecorateDeleteItem(
+	dec func(*dynamodb.DeleteItemInput) []func(*dynamodb.Options),
+) interface {
+	HandlerOption
+	ResourceRepositoryOption
+} {
+	return &options{
+		applyOptionToAdaptorFunc: func(d *decorators) {},
+		applyResourceRepositoryOptionFunc: func(d *decorators) {
+			d.decorateDeleteItem = dec
+		},
+	}
+}
+
+// WithDecorateTransactWriteItems adds a decorator for TransactWriteItems
+// operation. The decorator can modify the passed TransactWriteItemsInput
+// structure and return a slice of request.Option to alter the request prior to
+// its execution.
+//
+// Warning! Changing the order of transaction items in the original request may
+// lead to unpredictable results and errors. The logic of the projection handler
+// relies on the order of transaction items to distinguish between projection
+// OCC conflict and conditional failures of the transaction items provided by
+// the user.
+func WithDecorateTransactWriteItems(
+	dec func(*dynamodb.TransactWriteItemsInput) []func(*dynamodb.Options),
+) interface {
+	HandlerOption
+	ResourceRepositoryOption
+} {
+	return &options{
+		applyOptionToAdaptorFunc: func(d *decorators) {},
+		applyResourceRepositoryOptionFunc: func(d *decorators) {
+			d.decorateTransactWriteItems = dec
+		},
+	}
+}
+
+// WithDecorateCreateTable adds a decorator for CreateTable operation. The
+// decorator can modify the passed CreateTableInput structure and return
+// a slice of request.Option to alter the request prior to its execution.
```

Comment:
Please see suggested format on WithDecorateGetItem.

Reply:
Should be fixed in [5b85c8c](https://github.com/dogmatiq/projectionkit/pull/135/commits/5b85c8cd23a15394ccf4a704e5f4b3b26a2800fc)

Reply:
Should be fixed in [5b85c8c](https://github.com/dogmatiq/projectionkit/pull/135/commits/5b85c8cd23a15394ccf4a704e5f4b3b26a2800fc)

Reply:
Should be fixed in [Improve doc comments for option.](https://github.com/dogmatiq/projectionkit/pull/135/commits/6802c47336f4f20fee46481f6c4b17aa192b231e)